ReactAndroid/src/main/java/com/facebook/react/views/text/ReactTypefaceUtils.java

```diff
@@ -81,34 +84,46 @@ public static Typeface applyStyles(
       AssetManager assetManager) {
     int oldStyle;
     if (typeface == null) {
-      oldStyle = 0;
+      oldStyle = Typeface.NORMAL;
     } else {
       oldStyle = typeface.getStyle();
     }
 
-    int want = 0;
-    if ((weight == Typeface.BOLD)
-        || ((oldStyle & Typeface.BOLD) != 0 && weight == ReactTextShadowNode.UNSET)) {
-      want |= Typeface.BOLD;
+    int newStyle = oldStyle;
+    boolean italic = false;
+    if(weight == UNSET) weight = Typeface.NORMAL;
+    if(style == Typeface.ITALIC) italic = true;
+    boolean UNDER_SDK_28 = Build.VERSION.SDK_INT < Build.VERSION_CODES.P; 
+    boolean applyNumericValues = !(weight < (Typeface.BOLD_ITALIC + 1) || family != null);
+    boolean numericBold = UNDER_SDK_28 && weight > 699 && applyNumericValues;
+    boolean numericNormal = UNDER_SDK_28 && weight < 700 && applyNumericValues;
+    if (weight == Typeface.BOLD) {
+      newStyle = (newStyle == Typeface.ITALIC) ? Typeface.BOLD_ITALIC : Typeface.BOLD;
+      typeface = Typeface.create(typeface, newStyle);
+    } 
+    if (weight == Typeface.NORMAL) {
+      typeface = Typeface.create(typeface, Typeface.NORMAL);
+      newStyle = Typeface.NORMAL;
     }
-
-    if ((style == Typeface.ITALIC)
-        || ((oldStyle & Typeface.ITALIC) != 0 && style == ReactTextShadowNode.UNSET)) {
-      want |= Typeface.ITALIC;
+    if (style == Typeface.ITALIC) {
+      newStyle = (newStyle == Typeface.BOLD) ? Typeface.BOLD_ITALIC : Typeface.ITALIC;
+      typeface = Typeface.create(typeface, newStyle);
+    }
+    if(Build.VERSION.SDK_INT > Build.VERSION_CODES.O_MR1 && weight > Typeface.BOLD_ITALIC) {
+      typeface = Typeface.create(typeface, weight, italic);
+    }
+    if(family != null && UNDER_SDK_28 && weight > Typeface.BOLD_ITALIC) {
+      FLog.w(TAG, "Support for numeric font weight numeric values with custom fonts under Android API 28 Pie is not yet supported in ReactNative.");
     }
-
     if (family != null) {
-      typeface = ReactFontManager.getInstance().getTypeface(family, want, weight, assetManager);
-    } else if (typeface != null) {
-      // TODO(t9055065): Fix custom fonts getting applied to text children with different style
-      typeface = Typeface.create(typeface, want);
+      typeface = ReactFontManager.getInstance().getTypeface(family, newStyle, weight, assetManager);
     }
-
-    if (typeface != null) {
-      return typeface;
-    } else {
-      return Typeface.defaultFromStyle(want);
+    if (numericBold || numericNormal) {
+      newStyle = numericBold ? Typeface.BOLD : Typeface.NORMAL;
+      typeface = Typeface.create(typeface, newStyle);
+      FLog.w(TAG, "Support for numeric font weight numeric values available only from Android API 28 Pie. Android device lower then API 28 will use normal or bold.");
```

Comment:
I will change the logs to `FLog.d` as seems more appropriate for this use case

Reply:
I went ahead and made this change for you (in addition to formatting the file to conform to our lint rules). The merged commit should include this without any additional work from you. Thanks again!

Reply:
Thanks a lot @yungsters and sorry for not following up on this. I forgot about this comment. :pray: